packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ModelProvider.cs

```diff
@@ -341,6 +362,15 @@ private MethodBodyStatement GetPropertyInitializers(
                 return null;
             }
 
+            // validate if the additional properties property exists & if its' value type is also BinaryData
```

Comment:
Is the idea that we don't need the raw data field when there is an additional properties property?

Reply:
Not exactly. I took this condition from [autorest](https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProvider.cs#L76) since it looks like we don't generate the raw data field if the additional properties type is also `Dictionary<string, BinaryData>`. My assumption is because any additional raw data could just be stored in that additional properties property in that case.

Reply:
Ah, can we add a comment to that effect?